driver/registry_default.go

```diff
@@ -661,12 +669,25 @@ func (m *RegistryDefault) Courier(ctx context.Context) (courier.Courier, error)
 }
 
 func (m *RegistryDefault) ContinuityManager() continuity.Manager {
-	if m.continuityManager == nil {
+	// If m.continuityManager is nil or not a continuity.ManagerCookie
+	switch m.continuityManager.(type) {
+	case *continuity.ManagerCookie:
+	default:
 		m.continuityManager = continuity.NewManagerCookie(m)
 	}
 	return m.continuityManager
 }
 
+func (m *RegistryDefault) RelayStateContinuityManager() continuity.Manager {
+	// If m.continuityManager is nil or not a continuity.ManagerRelayState
+	switch m.continuityManager.(type) {
+	case *continuity.ManagerRelayState:
+	default:
+		m.continuityManager = continuity.NewManagerRelayState(m, m)
+	}
+	return m.continuityManager
+}
```

Comment:
This looks incorrect and racy. If you need two separate managers, save them in separate variables

Reply:
We don't need separate managers anymore, so problem solved :)